pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,251 @@
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
+	}
+
+	o.orgAccountListCacheTTL = t
+
+	return nil
+}
+
+func (o *orgValidator) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// IsMemberAccount method checks if the Account ID attached on the node is part of the organisation.
+// If it part of the organisation then validation should be succesfull if not attestation should fail, on enabling this verification method.
+// This could be alternative for not explicitly maintaining allowed list of account ids.
+// Method pulls the list of accounts from organization and caches it for certain time, cache time can be configured.
+func (o *orgValidator) IsMemberAccount(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDOfNode string) (bool, error) {
+	reValidatedcache, err := o.validateCache(ctx, orgClient)
+	if err != nil {
+		return false, err
+	}
+
+	accountIsmemberOfOrg, err := o.lookupCache(ctx, orgClient, accoundIDOfNode, reValidatedcache)
+	if err != nil {
+		return false, err
+	}
+
+	return accountIsmemberOfOrg, nil
+}
+
+// validateCache validates cache and refresh if its stale
+func (o *orgValidator) validateCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (bool, error) {
+	isStale := o.checkIfOrgAccountListIsStale(ctx)
+	// refresh the account map
+	if isStale {
+		_, err := o.reloadAccountList(ctx, orgClient, false)
+		if err != nil {
+			return true, err
+		}
+	}
+	return false, nil
+}
+
+func (o *orgValidator) lookupCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDOfNode string, reValidatedcache bool) (bool, error) {
+	o.mutex.RLock()
+	orgAccountList := o.orgListAccountMap
+	o.mutex.RUnlock()
+
+	_, accoutIsmemberOfOrg := orgAccountList[accoundIDOfNode]
+
+	// Retry if it doesn't exist in cache and cache was not revalidated
+	if !accoutIsmemberOfOrg && !reValidatedcache {
+		orgAccountList, err := o.refreshCache(ctx, orgClient)
+		if err != nil {
+			return false, err
+		}
+		_, accoutIsmemberOfOrg = orgAccountList[accoundIDOfNode]
+	}
+
+	return accoutIsmemberOfOrg, nil
+}
+
+// If cache miss happens, refresh list with new cache and check if element exist
+func (o *orgValidator) refreshCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (map[string]any, error) {
+	remTries := o.getRetries()
+
+	orgAccountList := make(map[string]any)
+	if remTries <= 0 {
+		return orgAccountList, nil
+	}
+
+	orgAccountList, err := o.reloadAccountList(ctx, orgClient, true)
+	if err != nil {
+		return nil, err
+	}
+
+	o.decrRetries()
+
+	return orgAccountList, nil
+}
+
+// checkIfOrgAccountListIsStale checks if the cached org account list is stale.
+func (o *orgValidator) checkIfOrgAccountListIsStale(_ context.Context) bool {
```

Comment:
if context is not used, may we remove that?
```suggestion
func (o *orgValidator) checkIfOrgAccountListIsStale() bool {
```